homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -137,32 +231,16 @@ def icon(self):
     def device_state_attributes(self):
         """Return the state attributes."""
         if self._type == "inverters":
-            return {"last_reported": self._last_reported}
-
-        return None
-
-    async def async_update(self):
-        """Get the energy production data from the Enphase Envoy."""
-        if self._type != "inverters":
-            _state = await getattr(self._envoy_reader, self._type)()
-            if isinstance(_state, int):
-                self._state = _state
-            else:
-                _LOGGER.error(_state)
-                self._state = None
-
-        elif self._type == "inverters":
+            serial_number = self._name.split(" ")[(len(self._name.split(" ")) - 1)]
             try:
-                inverters = await (self._envoy_reader.inverters_production())
-            except requests.exceptions.HTTPError:
-                _LOGGER.warning(
-                    "Authentication for Inverter data failed during update: %s",
-                    self._envoy_reader.host,
+                value = self.coordinator.data.get("inverters_production").get(
+                    serial_number
+                )[1]
+                return {"last_reported": value}
```

Comment:
Please only wrap the line that can raise in the try... except block. Move this down to the end.

Reply:
Removed try/except block from other comment.